static_frame/core/index.py

```diff
@@ -134,8 +134,24 @@ def mutable_immutable_index_filter(
         return index._MUTABLE_CONSTRUCTOR(index)
     return index.__class__(index) # create new instance
 
+
 #-------------------------------------------------------------------------------
 
+class _ArgsortCache(tp.NamedTuple):
+    arr: np.ndarray
+    key: np.ndarray
+
+    def __deepcopy__(self, memo: tp.Dict[int, tp.Any]) -> '_ArgsortCache':
+        obj = self.__class__.__new__(
+                self.__class__,
+                array_deepcopy(self.arr, memo=memo),
+                array_deepcopy(self.key, memo=memo),
+                )
```

Comment:
Do we need to call `__new__()` here?

```suggestion
        obj = self.__class__(
                array_deepcopy(self.arr, memo=memo),
                array_deepcopy(self.key, memo=memo),
                )
```